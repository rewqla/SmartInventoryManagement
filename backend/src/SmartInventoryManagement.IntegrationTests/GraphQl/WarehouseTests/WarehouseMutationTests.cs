using HotChocolate;
using HotChocolate.Execution;
using Snapshooter.Xunit;

namespace SmartInventoryManagement.IntegrationTests.GraphQl.WarehouseTests;

public class WarehouseMutationTests: IClassFixture<GraphQLServiceSetup>
{
    private readonly GraphQLServiceSetup _graphQlServiceSetup;

    public WarehouseMutationTests(GraphQLServiceSetup graphQlServiceSetup)
    {
        _graphQlServiceSetup = graphQlServiceSetup;
    }

    [Fact]
    public async Task GetWarehouses_FilterByLocation_ReturnsMatchingResults()
    {
        // Arrange & Act
        IExecutionResult result = await _graphQlServiceSetup.RequestExecutor.ExecuteAsync(
            @"
                mutation{
                  createWarehouse(input:  {
                         name: ""Normal Warehouse"",
                         location: ""Lutsk""
                      }){
                      name
                      location
                      __typename
                      errors{
                        ...on ValidationError{
                          message
                          errors{
                            propertyName
                            errorMessage
                          }
                        }
                      }
                    }
                  }");

        // Assert
        result.ToJson().MatchSnapshot();
    }
}
