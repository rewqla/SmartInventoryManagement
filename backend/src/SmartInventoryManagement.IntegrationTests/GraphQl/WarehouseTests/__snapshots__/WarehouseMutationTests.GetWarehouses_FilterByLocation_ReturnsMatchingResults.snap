{
  "data": {
    "createWarehouse": {
      "name": "Normal Warehouse",
      "location": "Lutsk",
      "__typename": "CreateWarehousePayload",
      "errors": null
    }
  }
}
